### PR TITLE
fix(scripts): issues with configuring typescript

### PR DESCRIPTION
**Pull request by SimeonC**
Small issues that cropped up during actual installation of projects to do with paths to node_modules and working directories. Discovered when setting up tablekit-v3.

Small feature added to the precommit to validate lerna sub-package versions all match up.

CleanShot 2022-07-06 at 12 43 50

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @tablecheck/codemods@1.2.0-canary.68.2627312741.0
  npm install @tablecheck/commitlint-config@1.2.0-canary.68.2627312741.0
  npm install @tablecheck/eslint-config@1.8.0-canary.68.2627312741.0
  npm install @tablecheck/scripts@2.3.0-canary.68.2627312741.0
  # or 
  yarn add @tablecheck/codemods@1.2.0-canary.68.2627312741.0
  yarn add @tablecheck/commitlint-config@1.2.0-canary.68.2627312741.0
  yarn add @tablecheck/eslint-config@1.8.0-canary.68.2627312741.0
  yarn add @tablecheck/scripts@2.3.0-canary.68.2627312741.0
  ```
</details>

### CODE REVIEW

File: packages/scripts/scripts/precommit.js [hunk 1]
```diff
@@ -23,20 +24,20 @@ async function runPrecommit() {
...
+
   const lintStagedConfig = {
     '**/*.{ts,tsx,js,jsx}': [
       // eslint fix first, otherwise eslint fix may unprettify files
       // also inherently checks typescript
-      'eslint --fix --no-eslintrc --config ./.eslintrc.js',
+      'eslint --fix --config ./.eslintrc.js',
       'prettier --write'
     ],
     '**/!(*.ts|*.tsx|*.js|*.jsx|package-json.json)': 'prettier --write -u',
```
Comment (staevs) [inline packages/scripts/scripts/precommit.js:43]: Would not this fixes be applied during eslint fix stage for ts*/js* files?

Comment (SimeonC) [reply packages/scripts/scripts/precommit.js:43]: No, we only have the eslint-config-prettier in the rules which doesn't fix anything. It seems like the [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) would fix this but we don't have it as part of eslint (it didn't exist when we setup this project)

File: packages/scripts/config/paths.js [hunk 2]
```diff
@@ -6,10 +6,24 @@ const fs = require('fs-extra');
...
 const resolveApp = (relativePath) => path.resolve(appDirectory, relativePath);
+// in this repo postinstall the env var is undefined
+const execLocation = process.env._ || './bin/postinstall.js';
+let systemDir = execLocation;
+const appRelativeSystemDir = path.join(
+  appDirectory,
+  'node_modules/@tablecheck/scripts'
+);
+if (fs.existsSync(appRelativeSystemDir)) systemDir = appRelativeSystemDir;
+// check for running itself - in npm postinstall or precommit hook here
```
Comment (gazpachu) [inline packages/scripts/config/paths.js:17]: @SimeonC Could we have a chat during the team meeting about the PR and react-system in general? I'm getting a bit lost

Comment (SimeonC) [reply packages/scripts/config/paths.js:17]: Added to the meeting agenda.

Comment (gazpachu) [reply packages/scripts/config/paths.js:17]: Thanks!

**Review by staevs — APPROVED** [review]

**Review by gazpachu — APPROVED** [review]

**Pull request merged**